frontend/src/metabase/browse/components/BrowseModels.tsx

```diff
@@ -0,0 +1,192 @@
+import _ from "underscore";
+import cx from "classnames";
+import { t } from "ttag";
+
+import type { Card, Collection, SearchResult } from "metabase-types/api";
+import * as Urls from "metabase/lib/urls";
+
+import Link from "metabase/core/components/Link";
+import LoadingAndErrorWrapper from "metabase/components/LoadingAndErrorWrapper";
+
+import type { useSearchListQuery } from "metabase/common/hooks";
+
+import { Box, Group, Icon, Text, Title } from "metabase/ui";
+import { sortModels } from "metabase/browse/utils";
+import NoResults from "assets/img/no_results.svg";
+import { useSelector } from "metabase/lib/redux";
+import { getLocale } from "metabase/setup/selectors";
+import { CenteredEmptyState } from "./BrowseApp.styled";
+import {
+  CollectionHeaderContainer,
+  CollectionHeaderLink,
+  GridContainer,
+  ModelCard,
+  MultilineEllipsified,
+} from "./BrowseModels.styled";
+import { LastEdited } from "./LastEdited";
+
+const emptyArray: SearchResult[] = [];
+
+export const groupModels = (
+  models: SearchResult[],
+  locale: string | undefined,
+) => {
+  const groupedModels = _.groupBy(models, model => model.collection.id);
+  let collections = models.map(model => model.collection);
+  collections = _.uniq(collections, collection => collection.id) || [];
+  collections.sort((a, b) => a.name?.localeCompare(b.name, locale));
+  return { groupedModels, collections };
```

Comment:
`let` is often a red flag for me. I think this would be simpler if we just sorted the grouped models and returned them as an array, then we wouldn't need to return collections at all.

```suggestion
  const groupedModels = Object.values(_.groupBy(models, model => model.collection.id))
    .sort((a, b) => a[0].collection?.name?.localeCompare(b[0].collection?.name, locale));
  return groupedModels;
```